File: Kth Largest Element in a Stream/Leetcode_703.cpp
class KthLargest {
private:
    int K;
    priority_queue<int> pq;
    
public:
    KthLargest(int k, vector<int>& nums) {
        K = k;
        for (auto it=nums.begin(); it!=nums.end(); ++it){
            pq.push(-*it);
            if (pq.size() > K) pq.pop();
        }
    }
    
    int add(int val) {
        pq.push(-val);
        if (pq.size()>K) pq.pop();
        return -pq.top();
    }
};

/**
 * Your KthLargest object will be instantiated and called as such:
 * KthLargest* obj = new KthLargest(k, nums);
 * int param_1 = obj->add(val);
 */

File: Kth Largest Element in a Stream/Leetcode_703.py
from heapq import heappop, heappush

class KthLargest:

    def __init__(self, k: int, nums: List[int]):
        self.k = k
        self.pq = []
        for num in nums:
            heappush(self.pq, num)
            if len(self.pq)>self.k: heappop(self.pq)  

    def add(self, val: int) -> int:
        heappush(self.pq, val)
        if len(self.pq)>self.k: heappop(self.pq)
        return self.pq[0]        


# Your KthLargest object will be instantiated and called as such:
# obj = KthLargest(k, nums)
# param_1 = obj.add(val)
